sending/base.py

```diff
@@ -68,15 +68,25 @@ def __init__(self):
         if not hasattr(self, "inbound_message_hook"):
             # Called by _inbound_worker when picking up a message from inbound queue
             # Primarily used for deserializing messages from the wire
+            # Will get one argument: incoming "raw" message content over the wire
```

Comment:
Could be useful to use `typing.Protocol` for these:

```python
class InboundMessageHook(typing.Protocol):
    async def __call__(self, raw: bytes):
        ...
````

Reply:
neat, will look into this for future upgrades. 🙇‍♂️